.pre-commit-config.yaml

```diff
@@ -18,6 +18,26 @@ repos:
     language: system
     files: >-
       ^setup\.py$
+  - id: check-python-packaging
+    name: Check python packaging
+    description: |
+        Performs validations of python packaging using both PEP-517 and old
+        setup.py while also validating metadata. Also verifies that the package
+        can be insalled in editable mode.
+    entry: >
+      bash -c "
+      rm -rf $VIRTUAL_ENV/dist.old $VIRTUAL_ENV/dist.new &&
+      python setup.py sdist bdist_wheel --dist-dir $VIRTUAL_ENV/dist.old &&
+      python -m twine check $VIRTUAL_ENV/dist.old/* &&
+      python -m pep517.build --source --binary --out-dir $VIRTUAL_ENV/dist.new . &&
+      python -m twine check $VIRTUAL_ENV/dist.new/* &&
+      python -m pip install -e ."
+    language: python
+    pass_filenames: false
+    always_run: true
```

Comment:
I don't think it's necessary. It should only run on changes to known packaging-related files.